### ISSUE TITLE
Error: NMF::nmf - Input matrix x contains at least one null or NA-filled row.

### ISSUE THREAD

**Issue by SamGG** [renozao/NMF#137]
Seems to occur when all the values in one row are zero. The message should reflect this if I am correct.

**Comment by ShixiangWang** [comment]
Yes, you can see the discussion at https://github.com/renozao/NMF/issues/6.

The solution is adding a small value to this matrix, like 0.001.

**Comment by elisagdelope** [comment]
The error message is not clear enough. It should clearly state all-zeroes row rather than, or in addition to, null or NA. 

In addition, I'm curious why NMF throws error in this case. In theory NMF can be performed even if there are rows in the input matrix that contain all zero values. I know there are implications of these all-zero rows on the factorization process and the interpretation of the results, but perhaps a warning would suffice as the operation is still feasible.
